packages/replayio/src/utils/recordings/upload/uploadRecordings.ts

```diff
@@ -30,7 +33,26 @@ export async function uploadRecordings(
 
   const multiPartUpload = await getFeatureFlagValue<boolean>("cli-multipart-upload", false);
   const client = new ProtocolClient();
-  await client.waitUntilAuthenticated();
+  try {
+    await client.waitUntilAuthenticated();
+  } catch (error) {
+    if (
+      error instanceof ProtocolError &&
+      error.protocolCode === AUTHENTICATION_REQUIRED_ERROR_CODE
```

Comment:
We likely should handle this on more places but I want to finish some other stuff right now and I didn't have time to investigate all of the places where it could be handled similarly. The CLI uses a mix of HTTP and WS requests~ so at times this stuff just errors differently between commands